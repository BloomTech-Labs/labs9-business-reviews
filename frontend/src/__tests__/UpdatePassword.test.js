import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import React from 'react';
import UpdatePasswordComponent from '../components/UpdatePassword';

describe('UpdatePasswordComponent />', () => {
  it('renders', () => {
    shallow(<UpdatePasswordComponent />);
  });
  it('matches snapshot', () => {
    const wrapper = shallow(<UpdatePasswordComponent />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
  it('should contain NavBar', () => {
    let app = shallow(<UpdatePasswordComponent />);
    expect(app.find("NavBar").length).toBe(1);
  });
  it('should contain two div elements', () => {
    let app = shallow(<UpdatePasswordComponent />);
    expect(app.find("div").length).toBe(2);
  });
  it('should contain one form', () => {
    let app = shallow(<UpdatePasswordComponent />);
    expect(app.find("form").length).toBe(1);
  });
  it('should contain two label elements', () => {
    let app = shallow(<UpdatePasswordComponent />);
    expect(app.find("label").length).toBe(2);
  });
  it('should contain three input elements', () => {
    let app = shallow(<UpdatePasswordComponent />);
    expect(app.find("input").length).toBe(3);
  });
});

