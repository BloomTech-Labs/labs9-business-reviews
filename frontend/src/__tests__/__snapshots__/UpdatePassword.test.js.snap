// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdatePasswordComponent /> matches snapshot 1`] = `
<div>
  <NavBar />
  <styled.div>
    <div
      className="image-container"
    >
      <h1>
         Hey 
         wanna reset your password?
      </h1>
    </div>
    <form
      onSubmit={[Function]}
    >
      <label>
        Password
      </label>
      <input
        name="password"
        onChange={[Function]}
        type="text"
        value=""
      />
      <label>
        Confirm Password
      </label>
      <input
        name="confirm_password"
        onChange={[Function]}
        type="text"
        value=""
      />
      <input
        type="submit"
        value="update password"
      />
    </form>
  </styled.div>
</div>
`;
